fix: allow react v 18 as peer dep

Thank you for taking the time to report an issue with react-d3-tree!  

Feel free to delete any questions that do not apply.

## Are you reporting a bug, or opening a feature request?
bug
  
## What is the actual behavior/output?
it show error while npm install

## What is the behavior/output you expect?
clear these the error while npm install

## Can you consistently reproduce the issue/create a reproduction case (e.g. on https://codesandbox.io)?
no

## What version of react-d3-tree are you using?
3.3.2

## If react-d3-tree crashed with a traceback, please paste the full traceback below.
npm ERR! Could not resolve dependency:
npm ERR! peer react@"16.x || 17.x" from react-d3-tree@3.3.2
npm ERR! node_modules/react-d3-tree
npm ERR!   react-d3-tree@"^3.3.2" from the root project

--- a/package.json
+++ b/package.json
@@ -73,7 +73,7 @@
     "uuid": "^8.3.1"
   },
   "peerDependencies": {
-    "react": "16.x || 17.x || ^18",
+    "react": "16.x || 17.x || 18.x",
     "react-dom": "16.x || 17.x || ^18"
   },
   "devDependencies": {

--- a/package.json
+++ b/package.json
@@ -74,7 +74,7 @@
   },
   "peerDependencies": {
     "react": "16.x || 17.x || 18.x",
-    "react-dom": "16.x || 17.x || ^18"
+    "react-dom": "16.x || 17.x || 18.x"
   },
   "devDependencies": {
     "@babel/core": "^7.8.3",

--- a/package.json
+++ b/package.json
@@ -73,8 +73,8 @@
     "uuid": "^8.3.1"
   },
   "peerDependencies": {
-    "react": "16.x || 17.x",
-    "react-dom": "16.x || 17.x"
+    "react": "16.x || 17.x || ^18",
+    "react-dom": "16.x || 17.x || ^18"
   },
   "devDependencies": {
     "@babel/core": "^7.8.3",

